diff --git a/es/components/menu/src/menu.mjs b/es/components/menu/src/menu.mjs
index 727663900176698325a66b549fab060b101237c3..612034858b11c7ca86b25bdcee0ae97517742193 100644
--- a/es/components/menu/src/menu.mjs
+++ b/es/components/menu/src/menu.mjs
@@ -168,6 +168,7 @@ var Menu = defineComponent({
       }
     };
     const calcMenuItemWidth = (menuItem) => {
+      if (!(menuItem instanceof Element)) return 0
       const computedStyle = getComputedStyle(menuItem);
       const marginLeft = Number.parseInt(computedStyle.marginLeft, 10);
       const marginRight = Number.parseInt(computedStyle.marginRight, 10);
@@ -177,7 +178,7 @@ var Menu = defineComponent({
       var _a, _b;
       if (!menu.value)
         return -1;
-      const items2 = Array.from((_b = (_a = menu.value) == null ? void 0 : _a.childNodes) != null ? _b : []).filter((item) => item.nodeName !== "#comment" && (item.nodeName !== "#text" || item.nodeValue));
+      const items2 = Array.from((_b = (_a = menu.value) == null ? void 0 : _a.childNodes) != null ? _b : []);
       const moreItemWidth = 64;
       const computedMenuStyle = getComputedStyle(menu.value);
       const paddingLeft = Number.parseInt(computedMenuStyle.paddingLeft, 10);
@@ -191,7 +192,7 @@ var Menu = defineComponent({
           sliceIndex2 = index + 1;
         }
       });
-      return sliceIndex2 === items2.length ? -1 : sliceIndex2;
+      return calcWidth <= menuWidth ? -1 : sliceIndex2;
     };
     const getIndexPath = (index) => subMenus.value[index].indexPath;
     const debounce = (fn, wait = 33.34) => {
