diff --git a/lib/module-factory.js b/lib/module-factory.js
index 34057e086b446e12de86ed02a457acd0c427b1a3..859cc51462c8ff4aa0a77a051558f4a702791895 100644
--- a/lib/module-factory.js
+++ b/lib/module-factory.js
@@ -133,7 +133,9 @@ var VuexClassModuleFactory = /** @class */ (function () {
                         _a)
                 });
             }
-            else {
+            else if(!store.hasModule(this.registerOptions.name)){
+                store.registerModule(this.registerOptions.name, vuexModule, {preserveState: true});
+            }else{
                 throw Error("[vuex-class-module]: A module with name '" + name + "' already exists.");
             }
         }